django/contrib/auth/forms.py

```diff
@@ -100,10 +100,17 @@ def clean_password2(self):
                 self.error_messages['password_mismatch'],
                 code='password_mismatch',
             )
-        self.instance.username = self.cleaned_data.get('username')
-        password_validation.validate_password(self.cleaned_data.get('password2'), self.instance)
         return password2
 
+    def _post_clean(self):
+        super()._post_clean()  # creates self.instance
```

Comment:
For the comment, I'd say `# updates self.instance with form data` (since `self.instance` is first assigned in ModelForm.__init__()` "creates" might be confusing)